highlight/highlight
# Update CTA for traces empty state

## Summary

Updates the empty state CTA for traces so they are bit more specific and helpful. Here is what we currently show in every instance.

<img width="1270" alt="Screenshot 2024-05-16 at 4 02 57 PM" src="https://github.com/highlight/highlight/assets/308182/6a26d9fb-26a0-47da-8623-94d8fec7e02c">

This PR adds a new CTA if we know they have integrated. The content varies depending on whether they have a query string entered.

<img width="1276" alt="Screenshot 2024-05-16 at 1 01 16 PM" src="https://github.com/highlight/highlight/assets/308182/eafd048a-2c78-4ccb-bcba-faa106cdc421">

<img width="1272" alt="Screenshot 2024-05-16 at 1 01 31 PM" src="https://github.com/highlight/highlight/assets/308182/c770ec01-5174-4ab9-8615-d7dd6ad89399">

This PR also makes some updates to the `Callout` component so you don't have to pass so much boilerplate to use a custom icon. It also leverages the component for the empty states instead of creating our own.

## How did you test this change?

* Go to the traces page and enter a random query that you know won't produce any results
* Ensure the CTA is correct
* Remove the query and select a very specific time range that you know won't produce any results
* Ensure the CTA is correct
* Test on a project that has not integrated tracing
* Ensure the CTA is correct
* Click around the app to other places you know there `Callout` is used and make sure they look alright

## Are there any deployment considerations?

N/A - click-side only change.

## Does this work require review from our design team?

N/A
